projects/dsp-ui/src/lib/viewer/operations/add-value/add-value.component.spec.ts

```diff
@@ -0,0 +1,214 @@
+
+@Component({
+    selector: `dsp-int-value`,
+    template: ``
+  })
+class TestIntValueComponent implements OnInit {
+
+    @Input() mode;
+
+    @Input() displayValue;
+
+    form: object;
+
+    ngOnInit(): void {
+
+        this.form = new FormGroup({
+            test: new FormControl(null, [Validators.required])
+        });
+    }
+
+    getNewValue(): CreateValue {
+        const createIntVal = new CreateIntValue();
+
+        createIntVal.int = 123;
+
+        return createIntVal;
+    }
+
+    updateCommentVisibility(): void { }
+}
+
+/**
+ * Test host component to simulate parent component.
+ */
+@Component({
+    selector: `dsp-add-value-host-component`,
+    template: `
+      <dsp-add-value #testAddVal [resourcePropertyDefinition]="resourcePropertyDefinition" [parentResource]="readResource"></dsp-add-value>`
+  })
+class DspAddValueTestComponent implements OnInit {
+
+    @ViewChild('testAddVal') testAddValueComponent: AddValueComponent;
+
+    readResource: ReadResource;
+    resourcePropertyDefinition: ResourcePropertyDefinition;
+
+    ngOnInit() {
+
+        MockResource.getTestthing().subscribe(res => {
+            this.readResource = res;
+        });
+
+        this.resourcePropertyDefinition = this.readResource.entityInfo.getAllPropertyDefinitions()[9] as ResourcePropertyDefinition;
+    }
+
+}
+
+describe('AddValueComponent', () => {
+    let testHostComponent: DspAddValueTestComponent;
+    let testHostFixture: ComponentFixture<DspAddValueTestComponent>;
+
+    beforeEach(async(() => {
+
+        const valuesSpyObj = {
+            v2: {
+                values: jasmine.createSpyObj('values', ['createValue', 'getValue'])
+            }
+        };
+
+        TestBed.configureTestingModule({
+            imports: [
+                BrowserAnimationsModule,
+                MatIconModule,
+            ],
+            declarations: [
+                AddValueComponent,
+                DspAddValueTestComponent,
+                TestIntValueComponent
+            ],
+            providers: [
+                {
+                    provide: DspApiConnectionToken,
+                    useValue: valuesSpyObj
+                },
+                EventBusService
+            ]
+        })
+        .compileComponents();
+    }));
+
+    beforeEach(() => {
+        testHostFixture = TestBed.createComponent(DspAddValueTestComponent);
+        testHostComponent = testHostFixture.componentInstance;
+        testHostFixture.detectChanges();
+
+        expect(testHostComponent).toBeTruthy();
+    });
+
+    it('should choose the apt component for an integer value', () => {
+
+        expect(testHostComponent.testAddValueComponent).toBeTruthy();
+
+        expect(testHostComponent.resourcePropertyDefinition.objectType).toEqual('http://api.knora.org/ontology/knora-api/v2#IntValue');
+
+        expect(testHostComponent.testAddValueComponent.createValueComponent instanceof TestIntValueComponent).toBeTruthy();
```

Comment:
added in db26c56

Reply:
Maybe we should think about how to make this template logic more generic (so it can be reused among different components), as described in https://dasch.myjetbrains.com/youtrack/issue/DSP-267